DoctrineBundle.php

```diff
@@ -119,17 +119,21 @@ public function shutdown()
         // Clear all entity managers to clear references to entities for GC
         if ($this->container->hasParameter('doctrine.entity_managers')) {
             foreach ($this->container->getParameter('doctrine.entity_managers') as $id) {
-                if (!$this->container instanceof IntrospectableContainerInterface || $this->container->initialized($id)) {
-                    $this->container->get($id)->clear();
+                if ($this->container->getParameter('doctrine.orm.entity_managers.' . $id . '.clear_on_shutdown') === true) {
```

Comment:
Putting the whole service id inside the parameter id looks wrong to me.

And your never use the configuration to configure this parameter, so it won't work.